base/structure.py

```diff
@@ -905,7 +907,7 @@ def __getitem__(self, index):
         elif isinstance(index, six.string_types):
             res = self.by_name(index)
         elif isinstance(index, slice):
-            res = [self.__getitem__(i) for i in six.moves.range(self.parent.ptr.memqty)].__getitem__(index)
+            res = [self.__getitem__(i) for i in range(self.parent.ptr.memqty)].__getitem__(index)
```

Comment:
No reason to explicitly call `object.__getitem__` here, when the native operator should just be used.